source/poggles/framebuffer.cpp

```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+
+#include "poggles/framebuffer.h"
+
+auto poggles::useDefaultFramebuffer() -> void
+{
+  glBindFramebuffer(GL_FRAMEBUFFER, 0);
+}
+
+poggles::framebuffer::framebuffer(GLenum target)
+    : m_original_target(target)
+{
+  glBindFramebuffer(m_original_target, m_framebuffer_handle.value());
+}
+
+auto poggles::framebuffer::use() -> void
+{
+  glBindFramebuffer(m_original_target, m_framebuffer_handle.value());
+}
+
+auto poggles::framebuffer::addTexture(GLenum attachment,
+                                      poggles::texture const& tex) -> bool
+{
+  tex.bind();
+  glBindFramebuffer(GL_FRAMEBUFFER, m_framebuffer_handle.value());
+  // TODO: Make sure this correctly handles incompatible textures
+  glFramebufferTexture2D(GL_FRAMEBUFFER,
+                         attachment,
+                         tex.target(),
+                         static_cast<GLuint>(tex),
+                         0);  // Level might need to be changed
+
+  bool status = true;
+
+  GLenum FBO_STATUS = glCheckFramebufferStatus(GL_FRAMEBUFFER);
+  if (FBO_STATUS != GL_FRAMEBUFFER_COMPLETE) {
+    status = false;
+
+    switch (FBO_STATUS) {
+      case GL_FRAMEBUFFER_UNDEFINED:
+        std::cout << "GL_FRAMEBUFFER_UNDEFINED" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_ATTACHMENT:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_ATTACHMENT" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_MISSING_ATTACHMENT:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_MISSING_ATTACHMENT"
+                  << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_DRAW_BUFFER:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_DRAW_BUFFER" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_READ_BUFFER:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_READ_BUFFER" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_UNSUPPORTED:
+        std::cout << "GL_FRAMEBUFFER_UNSUPPORTED" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_LAYER_TARGETS:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_LAYER_TARGETS" << std::endl;
+        break;
+    }
+  }
```

Comment:
Do we not have a better way of handling/reporting errors in poggles? (genuine question)

Reply:
define "better". I chose to not have poggles  choose or depend on a logger. We could switch to having it use `tl::expected`, but I also wasn't sure if that was a good choice for general GIV usage. However, I don't think this library gets general GIV usage, so I might be designing for a target audience that doesn't exist.

Reply:
We use error callbacks in the other libraries, right? Maybe just returning false would be good enough and we could rely on those

Reply:
At the very least just throwing the name into std out isn't very helpful. An actual message along with the name of the error to std error would be my preference.

Reply:
Me and Jeremy discussed this irl. We're going to just not do any reporting. OpenGL error reporting is outside the scope of what poggles does.